### DISCUSSION TITLE
Access external hard drive?

### DISCUSSION THREAD

**Discussion by glennman** [wolpi/prim-ftpd#321] [Q&A]
I have installed this app on an Android TV box running Android TV 11.
I am hoping to be able to access the external hard drive plugged into the TV box from my Windows 11 PC.
I am having two problems that I hope someone can help with.

Sometimes when I open the app, the menubar is missing, so I cannot do anything.
I found the problem/solution to this.
If the app does not have file write permissions, the menubar is not displayed.
Enabling file write permissions makes the menubar appear again.
This is very strange behavior.
Please consider changing this so the menubar is always displayed,
and display a message if the permissions are not set.

It uses the internal drive as the start folder.
I tried to change the start folder to the external drive, but could not figure out how to do that.
Another ftp app makes it easy to do that, so I know it is possible.
I finally got a terminal app installed and used df to find the path to the external drive:
\storage\xxxxxxxxxxx
I was able to set the ftpd start folder to this, and when I connect from my PC,
it displays the correct folder.
In the directory tree listing, it is listed under the \storage folder of the internal drive.
However, when I go down a few directory levels then click the .. icon to go back up,
ftpd reverts back to the internal drive, not the external drive.
The internal drive has a \storage folder, but it is empty.
The ftpd app does not seem to handle external drives properly.

Also, I tried to transfer a few files from the PC to the android external drive,
but the transfer failed. It says it transferred a few bytes, then aborted.
Sorry, I don't have the exact error message.
The same thing happened when I tried another android ftp app,
so that is probably some sort of file permission issue in android,
not the ftp app.

**Comment by wolpi** [comment]
>Sometimes when I open the app, the menubar is missing, so I cannot do anything.
>I found the problem/solution to this.
>If the app does not have file write permissions, the menubar is not displayed.
>Enabling file write permissions makes the menubar appear again.
>This is very strange behavior.
>Please consider changing this so the menubar is always displayed,
>and display a message if the permissions are not set.

Yes, that is strange. It is not intended to behave like this. There are reports that recent changes regarding Day/Night theme show such behavior on some devices. There are already code changes in this regard.



>I tried to change the start folder to the external drive, but could not figure out how to do that.

Android has some specific idea of security regarding files. It is not intended that apps access external storage via usual filesystem API. Google invented SAF for that.
